spread-shellcheck

```diff
@@ -133,15 +140,74 @@ def checksection(data, env: Dict[str, str]):
                             stdout=subprocess.PIPE,
                             stdin=subprocess.PIPE,
                             shell=True)
-    stdout, _ = proc.communicate(input='\n'.join(script_data).encode('utf-8'), timeout=30)
+    stdout, _ = proc.communicate(input='\n'.join(script_data).encode('utf-8'), timeout=60)
     if proc.returncode != 0:
         raise ShellcheckRunError(stdout)
 
 
+class Cacher:
+    _instance = None
+    def __new__(cls):
+        if cls._instance is not None:
+            return cls._instance
+        instance = super().__new__(cls)
+        cls._instance = instance
+        instance._enabled = True
+        instance._lock = Lock()
+        instance._hit =0
+        instance._miss = 0
+        return instance
+
+    def disable(self):
+        logging.debug("caching is disabled")
+        self._enabled = False
+
+    @staticmethod
+    def _cache_path_for(digest):
+        prefix = digest[:2]
```

Comment:
I assume this strips the 0x ? maybe a comment if so?

Reply:
No, this strips the first byte, so we have a structure like under `~/.git/objects/<byte-prefix>/<sha>`, eg `~/.cache/spread-shellcheck/47/477bc706b01e0b95587c5c8e5f783b47c83782cefbde8254f1985ac0430aa611`

Reply:
Sorry, I'm a muppet - makes sense.